src/frontend/Ast_to_Mir.ml

```diff
@@ -472,14 +472,26 @@ let rec trans_stmt (declc : decl_context) (ts : Ast.typed_statement) =
   let swrap stmt = [{stmt; smeta}] in
   let mloc = smeta in
   match stmt_typed with
-  | Ast.Assignment
-      { assign_lhs=
-          { assign_identifier
-          ; assign_indices
-          ; assign_meta= {id_ad_level; id_type_; lhs_ad_level; lhs_type_; loc}
-          }
-      ; assign_rhs
-      ; assign_op } ->
+  | Ast.Assignment {assign_lhs; assign_rhs; assign_op} ->
+      let rec get_lhs_base = function
+        | {Ast.lval= Ast.LIndexed (l, _); _} -> get_lhs_base l
+        | l -> l
+      in
+      let lhs_base = get_lhs_base assign_lhs in
+      let assign_identifier =
+        match lhs_base.Ast.lval with
+        | LVariable s -> s
+        | _ -> failwith "This should never happen."
+      in
+      let id_ad_level = lhs_base.Ast.lmeta.Ast.ad_level in
+      let id_type_ = lhs_base.Ast.lmeta.Ast.type_ in
+      let lhs_type_ = assign_lhs.Ast.lmeta.type_ in
+      let lhs_ad_level = assign_lhs.Ast.lmeta.ad_level in
+      let rec get_lhs_indices = function
```

Comment:
Your changes look awesome! My one comment would be that I think it'd just end up being cleaner to have `get_lhs_base` return the base identifier and the base identifier metadata separately rather than deconstructing it later. And you could probably have it return the indices at that time as well.